numpy/core/defchararray.py

```diff
@@ -1005,6 +1103,11 @@ def lstrip(a, chars=None):
     return _vec_string(a_arr, a_arr.dtype, 'lstrip', (chars,))
 
 
+def _partition_dispatcher(a, sep):
+    return (a,)
+
+
+@array_function_dispatch(_partition_dispatcher)
```

Comment:
I wonder if it makes sense to use `lambda` in these simple cases

Reply:
Yeah... I think there's a strong case for saying that it if fits on a single line with `lambda`, use that instead of a separate dispatcher (see my latest commit).

The downside is that it can look a little dense, e.g.,
```python
@array_function_dispatch(lambda a, old, new, count=None: (a,))
def replace(a, old, new, count=None):
    ...
```

Reply:
@mhvk @hameerabbasi any opinions on the style here?

Reply:
Actually, let's move discussion to the master issue.

Reply:
Could line wrap weirdly as:
```
@array_function_dispatch(lambda
            a, old, new, count=None: (a,))
def replace(a, old, new, count=None):
    ...
```

Reply:
I do like this wrapped version (added to the example in the master issue):
```python
@array_function_dispatch(
    lambda a, old, new, count=None: (a,)
)
def replace(a, old, new, count=None):
    ...
```